prefect_utils/snowflake.py

```diff
@@ -194,3 +198,157 @@ def load_ga_data_to_snowflake(
+    of each file name specified in the `files` parameter. Typically you would include a trailing (/)
+    in `s3_url`.
+
+    Args:
+      date (str): Date of the data being loaded.
+      date_property (str): Date type property name in the variant `PROPERTIES` column.
+      sf_credentials (dict): Snowflake public key credentials in the format required by create_snowflake_connection.
+      sf_database (str): Name of the destination database.
+      sf_schema (str): Name of the destination schema.
+      sf_table (str): Name of the destination table.
+      sf_role (str): Name of the snowflake role to assume.
+      sf_warehouse (str): Name of the Snowflake warehouse to be used for loading.
+      sf_storage_integration_name (str): Name of the Snowflake storage integration to use. These are
+              configured in Terraform.
+      s3_url (str): Full URL to the S3 path containing the files to load.
+      sf_file_format (str, optional): Snowflake file format for the Stage. Defaults to 'JSON'.
+      files (List[str], optional): File paths relative to `s3_url`.
+      pattern (str, optional): Path pattern/regex to match S3 objects to copy. Defaults to `None`.
+      overwrite (bool, optional): Whether to overwrite existing data for the given date. Defaults to `False`.
+    """
+
+    if not files and not pattern:
+        raise signals.Fail('Either `files` or `pattern` must be specified to run this task.')
+
+    sf_connection = create_snowflake_connection(sf_credentials, sf_role, warehouse=sf_warehouse)
+
+    # Check for data existence for this date
+    try:
+        query = """
+        SELECT 1 FROM {table}
+        WHERE date(PROPERTIES:{date_property})='{date}'
+        """.format(
+            table=qualified_table_name(sf_database, sf_schema, sf_table),
+            date=date,
+            date_property=date_property,
+        )
+        cursor = sf_connection.cursor()
+        cursor.execute(query)
+        row = cursor.fetchone()
+    except snowflake.connector.ProgrammingError as e:
+        if "does not exist" in e.msg:
+            # If so then the query failed because the table doesn't exist.
+            row = None
+        else:
+            raise
+
+    if row and not overwrite:
+        return
+
+    try:
+        # Create the generic loading table
+        query = """
+        CREATE TABLE IF NOT EXISTS {table} (
+            ID NUMBER AUTOINCREMENT START 1 INCREMENT 1,
+            LOAD_TIME TIMESTAMP_LTZ DEFAULT CURRENT_TIMESTAMP(),
+            ORIGIN_FILE_NAME VARCHAR(16777216),
+            ORIGIN_FILE_LINE NUMBER(38,0),
+            ORIGIN_STR VARCHAR(16777216),
+            PROPERTIES VARIANT
+        );
+        """.format(
+            table=qualified_table_name(sf_database, sf_schema, sf_table)
+        )
+
+        sf_connection.cursor().execute(query)
+
+        # Delete existing data in case of overwrite.
+        if overwrite:
```

Comment:
Not super necessary, but it would speed things up a bit to check `row` here and not execute the delete if there's nothing to delete. This might be more important as these tables grow huge. In fact, we may want to create a cluster key on `PROPERTIES:{date_property}` too?